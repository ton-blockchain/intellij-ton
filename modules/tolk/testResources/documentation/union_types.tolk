/// Function returning nullable type
fun maybeValue(): int? {
    return null;
}

/// Function with complex union
fun multiUnion(): int | slice | bool | null {
    return 42;
}

/// Simple union parameter
fun processUnion(value: slice | int): void {
}
