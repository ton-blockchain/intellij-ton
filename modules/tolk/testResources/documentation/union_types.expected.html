<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Tolk Documentation Test Results</title>
    <style>
        body { font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; margin: 20px; background: #fafafa; }
        h1 { color: #2c3e50; border-bottom: 3px solid #3498db; padding-bottom: 10px; }
        .element { margin-bottom: 25px; border: 1px solid #ddd; padding: 20px; background: white; border-radius: 8px; box-shadow: 0 2px 4px rgba(0,0,0,0.1); }
        .element-title { font-weight: bold; color: #2c3e50; margin-bottom: 15px; font-size: 14px; background: #ecf0f1; padding: 8px; border-radius: 4px; }
        .documentation { background: #f8f9fa; padding: 15px; border-radius: 6px; border-left: 4px solid #3498db; font-family: 'JetBrains Mono', 'Fira Code', monospace; line-height: 1.6; }
        
        /* Tolk syntax highlighting */
        .keyword { color: #d73a49; font-weight: bold; }
        .primitive { color: #005cc5; font-weight: 500; }
        .struct { color: #22863a; font-weight: 500; }
        .function { color: #6f42c1; font-weight: 500; }
        .parameter { color: #e36209; }
        .type-alias { color: #d73a49; }
        .type-parameter { color: #6f42c1; font-style: italic; }
        .annotation { color: #6a737d; }
        .string { color: #032f62; }
        .number { color: #005cc5; }
        .comment { color: #6a737d; font-style: italic; }
        .constant { color: #005cc5; font-weight: 500; }
        .global-variable { color: #e36209; font-weight: 500; }
        .identifier { color: #24292e; }
        
        /* Documentation markup */
        .definition-start { border-top: 2px solid #e1e4e8; margin-top: 10px; padding-top: 10px; }
        .definition-end { border-bottom: 1px solid #e1e4e8; margin-bottom: 10px; padding-bottom: 10px; }
        .content-start { margin-top: 15px; }
    </style>
</head>
<body>
    <h1>Tolk Documentation Test Results</h1>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int?</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span><span style="">?</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: maybeValue</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">maybeValue</span><span style="">()</span>: <span style="color:#000080;font-weight:bold;">int</span><span style="">?</span></pre></div><div class='content'><p>Function returning nullable type</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: slice</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">slice</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: bool</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">bool</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: null</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">null</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int | slice | bool | null</div>
        <div class="documentation">
            
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">int</span>
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">slice</span>
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">bool</span>
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">null</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: multiUnion</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">multiUnion</span><span style="">()</span>: 
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">int</span>
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">slice</span>
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">bool</span>
    <span style="">|</span> <span style="color:#000080;font-weight:bold;">null</span></pre></div><div class='content'><p>Function with complex union</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: slice</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">slice</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: slice | int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">slice</span> <span style="">|</span> <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Parameter: value</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000000;">value</span>: <span style="color:#000080;font-weight:bold;">slice</span> <span style="">|</span> <span style="color:#000080;font-weight:bold;">int</span></pre></div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: void</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">void</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Function: processUnion</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">fun</span> <span style="color:#000000;">processUnion</span><span style="">(</span><span style="color:#000000;">value</span>: <span style="color:#000080;font-weight:bold;">slice</span> <span style="">|</span> <span style="color:#000080;font-weight:bold;">int</span><span style="">)</span>: <span style="color:#000080;font-weight:bold;">void</span></pre></div><div class='content'><p>Simple union parameter</p>
</div>
        </div>
    </div>
</body>
</html>
